sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -104,6 +105,23 @@ abstract class JdbcDialect extends Serializable with Logging {
    */
   def getJDBCType(dt: DataType): Option[JdbcType] = None
 
+  /**
+   * Convert java.sql.Timestamp to a Long value (internal representation of a TimestampNTZType)
+   * holding the microseconds since the epoch of 1970-01-01 00:00:00Z for this timestamp.
+   */
+  def convertJavaTimestampToTimestampNTZ(t: Timestamp): Long = {
+    DateTimeUtils.fromJavaTimestampNoRebase(t)
+  }
+
+  /**
+   * Converts a LocalDateTime representing a TimestampNTZ type to an
+   * instance of `java.sql.Timestamp`.
+   */
+  def convertTimestampNTZToJavaTimestamp(ldt: LocalDateTime): Timestamp = {
```

Comment:
these two are not symmetry. Shall we let `convertJavaTimestampToTimestampNTZ` return `LocalDateTime` and then Spark converts `LocalDateTime` to a long?

Reply:
It would be hard to make them symmetric actually.
For example, the neutral conversion solution here stores the Java Timestamp's underlining microseconds from epoch directly into the long value. It would be wrong to construct a `LocalDateTime` from the year, day, ..., minute, second fields in the Timestamp.

Reply:
People can use `LocalDateTime.ofEpochSecond`, no need to specify year, month, day, ... fields